src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -237,6 +246,50 @@ RestExecuteOnExtensionResponse handleRestExecuteOnExtensionRequest(RestExecuteOn
                 "No handler for " + ExtensionRestPathRegistry.restPathToString(request.getMethod(), request.getUri())
             );
         }
+
+        // TODO get this list of resource IDs and addition params from the RestExecuteOnExtentionRequest
+        Map<String, Object> params = new HashMap<>();
+
+        // TODO Create a general mechanism for getting the name of the permission for an action
+        String requiredPermission = "deny_by_default";
+        if (restHandler instanceof CrudRestHandler) {
+            ProtectedRoute matchingRoute = ((CrudRestHandler) restHandler).getMatchingRoute(request.getMethod(), request.getUri());
+            if (matchingRoute != null) {
+                params = matchingRoute.getConsumedParams(request.getUri());
+                requiredPermission = matchingRoute.getRequiredPermission();
+            }
+        }
+
+        // TODO validate that caller has the permissions required here
+        PrincipalIdentifierToken requestIssuerIdentifier = request.getRequestIssuerIdentity();
+
+        AuthorizationResponseHandler authorizationResponseHandler = new AuthorizationResponseHandler();
+        try {
+            AuthorizationResponse authResponse = this.extensionTransportService.submitRequest(
+                    opensearchNode,
+                    ExtensionsOrchestrator.REQUEST_EXTENSION_AUTHORIZE_ACTION,
+                    new AuthorizationRequest(getUniqueId(), requestIssuerIdentifier, requiredPermission, params),
```

Comment:
`AuthorizationRequest` accepts String as inputType for requestIssueIdentifier (refer [here](https://github.com/opensearch-project/OpenSearch/pull/4516/files#diff-21e48b040c4cc6daff2bb4601378a730026ce338fdae278524e226894cc9aab1R32)), but we are passing in the PrincipalidentifierToken object which creates a mismatch. 

How should we proceed with this?

Reply:
The PR against core was opened to solicit feedback on the interface for submitting AuthorizationRequests from extensions. Passing a String in as a PrincipalIdentifierToken is a placeholder until the work to pass tokens to extensions is complete.

Reply:
okay..gotcha